### DISCUSSION TITLE
usage of create_async_engine().sync_engine

### DISCUSSION THREAD

**Discussion by cmflynn** [sqlalchemy/sqlalchemy#6605] [Usage Questions]
I'm not clear how to use the sync engine. I tried simply creating a sync session from the sync engine and using in sync code but that throws an error: `sqlalchemy.exc.MissingGreenlet: greenlet_spawn has not been called; can't call await_() here. Was IO attempted in an unexpected place? (Background on this error at: http://sqlalche.me/e/14/xd2s)
`

here's some sample code: 
```python
engine: Engine = create_async_engine(
    url, echo=True, future=True
)

async_session = sessionmaker(
    engine, class_=AsyncSession, expire_on_commit=False, future=True
)

sync_session = scoped_session(
    sessionmaker(
        autocommit=False,
        autoflush=False,
        bind=engine.sync_engine,
    ),
)

res = sync_session.query(SomeModel)
res = res.all()
```

Am i missing something?

Worth mentioning that I discovered sync_engine from another question around event handlers.

```python
event.listens_for(engine.sync_engine, "do_connect")(do_connect_listener)
event.listens_for(engine.sync_engine, "engine_connect")(engine_connect_listener)
```

**Comment by CaselIT** [comment]
Hi,

The async extension of sqlalchemy uses proxies to the normal sqlalchemy objects. In case of the engine, the proxied object is held in the sync_engine property. Publicly there is no api other than the events one where that's needed. To interact with it you should use the async version of the session.
You may use [run_sync](https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncSession.run_sync) if you need to access a session that can be used with a sync api.
In particular currently `scoped_session` is not supported in the async extension. see https://github.com/sqlalchemy/sqlalchemy/issues/6583 for details
